### PR TITLE
fix(payments): do not show error message on empty confirm email input 

### PR DESCRIPTION
**Pull request by IvoJP**
Because:

* We would like UI that only displays errors on user input

This commit:

* Hides the error state unless there is an error due to user input

Closes: #10448

## Checklist

_Put an `x` in the boxes that apply_

- [X] My commit is GPG signed.
- [X] If applicable, I have modified or added tests which pass locally.
- [X] I have added necessary documentation (if appropriate).
- [X] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx [hunk 1]
```diff
@@ -127,11 +127,17 @@ describe('NewUserEmailForm test', () => {
...
       fireEvent.change(firstEmail, { target: { value: 'valid@email.com' } });
-      fireEvent.change(secondEmail, {
-        target: { value: 'not.the.same@email.com' },
+
+      // need to wait for the first event on secondEmail before calling blur
+      // otherwise blur will discard the change event
+      await act(async () => {
+        fireEvent.change(secondEmail, {
+          target: { value: 'not.the.same@email.com' },
+        });
```
Comment (biancadanforth) [inline packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx:140]: I'm not sure why your `await act...` works (I'm guessing it just slows down the script enough for the async validation on the first input field to complete before firing the next event), but I think the actual root cause is that we're not waiting for the async validation from the first input field's change event to complete.

I tried this diff locally in your feature branch, and it passes -- perhaps this would be a more clear fix for this test failure? This would match similar test patterns we have where we mock (async) API calls.

```diff
diff --git a/packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx b/packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx
index 541e3f147..6a9d0e8cf 100644
--- a/packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx
+++ b/packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx
@@ -12,8 +12,13 @@ jest.mock('../../lib/apiClient', () => ({
 }));
 import { apiFetchAccountStatus } from '../../lib/apiClient';
 
-import { NewUserEmailForm, emailInputValidationAndAccountCheck } from './index';
+import {
+  NewUserEmailForm,
+  emailInputValidationAndAccountCheck,
+  checkAccountExists,
+} from './index';
 import { Localized } from '@fluent/react';
+import waitForExpect from 'wait-for-expect';
 const selectedPlan = {
   plan_id: 'planId',
   plan_name: 'Pro level',
@@ -53,10 +58,8 @@ const WrapNewUserEmailForm = ({
         setEmailsMatch={setEmailsMatch}
         setValidEmail={setValidEmail}
         setAccountExists={setAccountExists}
+        checkAccountExists={checkAccountExists}
         getString={(id: string) => id}
-        checkAccountExists={() =>
-          Promise.resolve({ exists: accountExistsReturnValue })
-        }
         selectedPlan={selectedPlan}
       />
     </div>
@@ -131,13 +134,11 @@ describe('NewUserEmailForm test', () => {
       const firstEmail = subject.getByTestId('new-user-email');
       const secondEmail = subject.getByTestId('new-user-confirm-email');
       fireEvent.change(firstEmail, { target: { value: 'valid@email.com' } });
-
-      // need to wait for the first event on secondEmail before calling blur
-      // otherwise blur will discard the change event
-      await act(async () => {
-        fireEvent.change(secondEmail, {
-          target: { value: 'not.the.same@email.com' },
-        });
+      await waitForExpect(() =>
+        expect(apiFetchAccountStatus).toHaveBeenCalledTimes(1)
+      );
+      fireEvent.change(secondEmail, {
+        target: { value: 'not.the.same@email.com' },
       });
       fireEvent.blur(secondEmail);
     });
```

**Review by biancadanforth — CHANGES_REQUESTED** [review]
Thanks Ivo for this PR! Everything looks pretty good -- I mostly just want to modify the test fix to be a little more clear in terms of what we're waiting on.

I make some other small, non-blocking suggestions as well.

File: packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx [hunk 2]
```diff
@@ -241,8 +241,8 @@ export function emailConfirmationValidation(
 
   return {
     value,
-    valid,
-    error: !valid && !focused ? errorMsg : null,
+    valid: valid || !value,
```
[the remark this answers is not in the document]
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx:244]: emailsMatch in the scope of where this defined is actually a helper method. the boolean emailsMatch that is out of scope but is updated by the function passed in as parameter to this component from the checkout component - thats how it gets updated even though its not directly accessible from here (hopefully that made sense).

File: packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx (hunk 1 above)
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx]: per the discussion in slack, we went with removing the wrapping act call on the body of the test - we think that might have been the main issue as the test is pretty straight forward and works as expected without that superfluous act.

let me know if this is satisfactory from your point of view.

[the remark this answers is not in the document]
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx]: so for this test, it should have the class invalid as this tests two explicitly different emails. i think what youre suggesting is another test where one email is provided and the second email is left blank?

let me know if you'd like me to add that test as part of this PR.

File: packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (biancadanforth) [reply packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx:244]: Hmm I'm not following you here. I am seeing that `emailsMatch` is a boolean state in the parent component, `Checkout`, but it isn't defined anywhere in `NewUserEmailForm`. Its value in the parent component is set when we call `setEmailsMatch`. Using the block-scoped variable `emailsMatch` here won't affect that, but you could change the variable name slightly also if you prefer?

File: packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (biancadanforth) [reply packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx]: Ah you're right; you're modifying an existing test, and we didn't have test coverage for this specific case of an empty string that you're covering in this PR.

Yes I think it's a good practice to write a new test case for the fixed behavior for a bug in addition to fixing any existing test cases that may have regressed. Can you add one?

Comment (biancadanforth) [reply packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx]: Hmm true that `render` is also already wrapped in `act` and is sync, just like `fireEvent`. If that works then that's great. I guess we don't actually need to wait for the async validation on the first email input here, since our second email input's validation is based on the value of the first input field, not the result of its async validation.

WFM thanks.

File: packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (biancadanforth) [reply packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx:244]: Either way, it's not blocking of course.

**Review by biancadanforth — APPROVED** [review]
R + WC. This looks good Ivo; can you add a new test case for the empty string case from the ticket here (and make sure CI passes again) before merging?

File: packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/NewUserEmailForm/index.test.tsx]: test case added

File: packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx:244]: it looks like an unused import from a helpers file at the top of `NewUserEmailForm`:  `import { emailsMatch, isEmailValid } from '../../../../fxa-shared/email/helpers'` which is a function not the boolean variable in the state of the parent component. I'll go ahead and remove the unused import so that `emailsMatch` (the function) isn't in scope anymore and it won't appear with intellisense in vscode.

the boolean `emailsMatch` in the parent component is never passed into `NewUserEmailForm` so its values in inaccessible to be read. The `NewUserEmailForm` takes a setState function which provides a closure around the variable for the child component to update but it'd require a refactor to allow this boolean to be read from the child component - i'm guessing thats why the original author did this `  let valid = false; setEmailsMatch((valid = emailInputState === value));` so that they could work with the same value without having to pass it into the child component.
  
let me know if you'd like me to go forward with that refactor.

[the remark this answers is not in the document]
Comment (IvoJP) [reply packages/fxa-payments-server/src/components/NewUserEmailForm/index.tsx:244]: also let me know if there is another way to access a parent components state without explicitly passing it into the child component. i havent done much work with react and its been a bit since i used vue.
